src/app/ui/editor/pixels_movement.cpp

```diff
@@ -495,9 +497,12 @@ void PixelsMovement::moveImage(const gfx::PointF& pos, MoveModifier moveModifier
 
       // Snap to grid when resizing tilemaps
       if (m_site.tilemapMode() == TilemapMode::Tiles) {
-        gfx::Rect gridBounds = m_site.gridBounds();
-        a = gfx::PointF(snap_to_grid(gridBounds, gfx::Point(a), PreferSnapTo::BoxOrigin));
-        b = gfx::PointF(snap_to_grid(gridBounds, gfx::Point(b), PreferSnapTo::BoxOrigin));
+        b.x--;
+        b.y--;
+        gfx::RectF alignedRect(m_site.grid().alignBounds(gfx::RectF(a, gfx::SizeF(1, 1)) |
+                                                         gfx::RectF(b, gfx::SizeF(1, 1))));
+        a = alignedRect.origin();
+        b = gfx::PointF(alignedRect.x2(), alignedRect.y2());
```

Comment:
This part, fixes the Scale manipulation on TilemapMode. Without this, the position will jump to other unexpected positions.